test/unit/expressions.spec.ts

```diff
@@ -18,8 +18,56 @@ test.each([
     util.testFunction(input).disableSemanticCheck().expectLuaToMatchSnapshot();
 });
 
-test.each(["3+4", "5-2", "6*3", "6**3", "20/5", "15/10", "15%3"])("Binary expressions basic numeric (%p)", input => {
-    util.testExpression(input).expectToMatchJsResult();
+const numericSupportedInAll = ["3+4", "5-2", "6*3", "6**3", "20/5", "15/10"];
+const translatedIn50 = ["15%3"];
+const allNumericOperators = [...numericSupportedInAll, ...translatedIn50];
+test.each(numericSupportedInAll)("Binary expressions basic numeric [5.0] (%p)", input => {
+    util.testExpression(input)
+        .setOptions({ luaTarget: tstl.LuaTarget.Lua50, luaLibImport: tstl.LuaLibImportKind.None })
+        .disableSemanticCheck()
+        .expectToMatchJsResult();
+});
+
+test.each(translatedIn50)("Translated binary expressions basic numeric [5.0] (%p)", input => {
+    util.testExpression(input)
+        .setOptions({ luaTarget: tstl.LuaTarget.Lua50, luaLibImport: tstl.LuaLibImportKind.None })
+        .disableSemanticCheck()
+        .expectLuaToMatchSnapshot();
+});
+
+test.each(allNumericOperators)("Binary expressions basic numeric [5.1] (%p)", input => {
+    util.testExpression(input)
+        .setOptions({ luaTarget: tstl.LuaTarget.Lua51, luaLibImport: tstl.LuaLibImportKind.None })
+        .disableSemanticCheck()
+        .expectToMatchJsResult();
+});
+
+test.each(allNumericOperators)("Binary expressions basic numeric [JIT] (%p)", input => {
+    util.testExpression(input)
+        .setOptions({ luaTarget: tstl.LuaTarget.LuaJIT, luaLibImport: tstl.LuaLibImportKind.None })
+        .disableSemanticCheck()
+        .expectToMatchJsResult();
+});
+
+test.each(allNumericOperators)("Binary expressions basic numeric [5.2] (%p)", input => {
+    util.testExpression(input)
+        .setOptions({ luaTarget: tstl.LuaTarget.Lua52, luaLibImport: tstl.LuaLibImportKind.None })
+        .disableSemanticCheck()
+        .expectToMatchJsResult();
+});
+
+test.each(allNumericOperators)("Binary expressions basic numeric [5.3] (%p)", input => {
+    util.testExpression(input)
+        .setOptions({ luaTarget: tstl.LuaTarget.Lua53, luaLibImport: tstl.LuaLibImportKind.None })
+        .disableSemanticCheck()
+        .expectToMatchJsResult();
+});
+
+test.each(allNumericOperators)("Binary expressions basic numeric [5.4] (%p)", input => {
+    util.testExpression(input)
+        .setOptions({ luaTarget: tstl.LuaTarget.Lua54, luaLibImport: tstl.LuaLibImportKind.None })
+        .disableSemanticCheck()
+        .expectToMatchJsResult();
```

Comment:
This seems way too complicated, seems to me these 50 lines could be replaced with just:
```ts
for (const expression of ["3+4", "5-2", "6*3", "6**3", "20/5", "15/10", "15%3"]) {
    util.testEachVersion(
        `Binary expressions basic numeric (${expression})`,
        () => util.testExpression(expression),
        util.expectEachVersionExceptJit(b => b.expectToMatchJsResult())
    );
}
```
All of these should be supported in all versions, and we want to minimize the use of snapshots so that can just be left out.

Reply:
Very well. Although I assume we can't execute any Lua 5.0 code until we add the wasm bindings for that version.

Reply:
Correct, unfortunately we cannot test that now, but a snapshot still wouldn't really test if that actually worked in 5.0 either

Reply:
Okay, I've adopted your approach with 555e2f855719e7830d446e857fc58d48dd57b00f.